src/item_location.cpp

```diff
@@ -478,6 +478,101 @@ class item_location::impl::item_on_vehicle : public item_location::impl
+{
+    private:
+        item_location container;
+
+        // figures out the index for the item, which is where it is in the total list of contents
+        // note: could be a better way of handling this?
+        int calc_index() const {
+            int idx = 0;
+            for( const item &it : container->contents.all_items() ) {
+                if( target() == &it ) {
+                    return idx;
+                }
+                idx++;
+            }
+            if( idx == 0 ) {
+                return -1;
+            }
+            return idx;
+        }
+    public:
+        item_in_container( const item_location &container, item *which ) :
+            impl( which ), container( container ) {}
+
+        void serialize( JsonOut &js ) const override {
+            js.start_object();
+            js.member( "idx", calc_index() );
+            js.member( "type", "in_container" );
+            js.member( "parent", container );
+            js.end_object();
+        }
+
+        item *unpack( int idx ) const override {
+            std::list<item> all_items = container->contents.all_items();
+            auto iter = all_items.begin();
+            std::advance( iter, idx );
+            if( iter != all_items.end() ) {
+                return &*iter;
+            } else {
+                return nullptr;
+            }
+        }
+
+        std::string describe( const Character * ) const override {
+            if( !target() ) {
+                return std::string();
+            }
+            return _( string_format( "inside %s", container->tname() ) );
+        }
+
+        type where() const override {
+            return type::container;
+        }
+
+        tripoint position() const override {
+            return container.position();
+        }
+
+        void remove_item() override {
+            container->contents.remove_item( *target() );
+        }
+
+        int obtain( Character &ch, int qty ) override {
+            ch.mod_moves( -obtain_cost( ch, qty ) );
+
+            item obj = target()->split( qty );
+            if( !obj.is_null() ) {
+                return ch.get_item_position( &ch.i_add( obj, should_stack ) );
+            } else {
+                int inv = ch.get_item_position( &ch.i_add( *target(), should_stack ) );
+                remove_item();
+                return inv;
+            }
+        }
+
+        int obtain_cost( const Character &ch, int qty ) const override {
+            if( !target() ) {
+                return 0;
+            }
+
+            item obj = *target();
+            obj = obj.split( qty );
+            if( obj.is_null() ) {
+                obj = *target();
+            }
+
+            const int container_mv = container->contents.obtain_cost( *target() );
+            if( container_mv == 0 ) {
+                debugmsg( string_format( "ERROR: %s does not contain %s", container->tname(), target()->tname() ) );
```

Comment:
```suggestion
                debugmsg( "ERROR: %s does not contain %s", container->tname(), target()->tname() );
```
`debugmsg` does string formatting.